### DISCUSSION TITLE
ModuleNotFoundError for custom modules when running executable

### DISCUSSION THREAD

**Discussion by jezztify** [pyinstaller/pyinstaller#6457] [Help] [answered]
I've searched the issues and discussions and have tried out some of the things that were suggested in different threads but still am not able to make it work. I'm new to using pyinstaller.

I have an open-source project that I want to use pyinstaller to create a distributable executable file.

https://github.com/jezztify/PoE-Pvt-League-Ranker-Overlay/tree/create_executable

TLDR: below is the file structure

```
root
  | - pplo
      | - __init__.py
      | - __main__.py
      | - overlay.py
      | - charData.py
      | - plData.py
  | - setup.py
  | - config.ini
[... some other files ...]
```

I have custom modules imported inside `__main__.py`.
I have created a spec file via `pyi-makespec` and below is the result

```
# -*- mode: python ; coding: utf-8 -*-


block_cipher = None


a = Analysis(['pplo\\__main__.py'],
             pathex=[],
             binaries=[],
             datas=[],
             hiddenimports=[],
             hookspath=[],
             hooksconfig={},
             runtime_hooks=[],
             excludes=[],
             win_no_prefer_redirects=False,
             win_private_assemblies=False,
             cipher=block_cipher,
             noarchive=False)
pyz = PYZ(a.pure, a.zipped_data,
             cipher=block_cipher)

exe = EXE(pyz,
          a.scripts, 
          [],
          exclude_binaries=True,
          name='__main__',
          debug=False,
          bootloader_ignore_signals=False,
          strip=False,
          upx=True,
          console=True,
          disable_windowed_traceback=False,
          target_arch=None,
          codesign_identity=None,
          entitlements_file=None )
coll = COLLECT(exe,
               a.binaries,
               a.zipfiles,
               a.datas, 
               strip=False,
               upx=True,
               upx_exclude=[],
               name='__main__')
```

when I run `pyinstaller --onefile __main__.spec pplo/__main__.py` it generates the executable but when I start running it, I end up with

```
Traceback (most recent call last):
  File "pplo\__main__.py", line 3, in <module>
ModuleNotFoundError: No module named 'overlay'
[18800] Failed to execute script '__main__' due to unhandled exception!
```

Am I missing anything? Thanks in advance.

**Comment by jezztify** [comment]
I kind of figured something out and below command worked
```
pyinstaller --noconfirm --onefile --console --clean --paths "/path/to/PoE-Pvt League Overlay/pplo"  "/path/to/PoE-Pvt League Overlay/pplo/__main__.py"
```
just one more thing is that, it doesn't include the `config.ini` and the `pplo/data` folder.

**Comment by jezztify** [comment, marked as answer]
just created a `build` binary as below

```
echo "----- Removing dist folder"
rm -rf dist
echo "----- Compiling binary"
pyinstaller --noconfirm --onefile --console --clean --paths "pplo"  "pplo/__main__.py"
echo "----- Copying files"
cp config.ini dist/
mkdir -p dist/pplo
cp -r pplo dist/
echo "----- Done!"
```

above works but if `pyinstaller` has a similar capability, I would be very happy to know :)

**Reply by rokm** [reply]
You mean something like [this](https://pyinstaller.readthedocs.io/en/stable/usage.html#cmdoption-add-data) or [this](https://pyinstaller.readthedocs.io/en/stable/usage.html#cmdoption-collect-data)?

**Reply by jezztify** [reply]
will check this. Thank you for pointing it out! :)

//edit

tried using `--add-data` but doesn't work.
